src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -528,30 +575,75 @@ def getMSOutputTransferInfo(self, wflow):
             wflow['TransferDone'] = True
         return wflow
 
+    def setParentDatasets(self, wflow):
+        """
+        Used to resolve parent datasets for a workflow.
+        :param  wflow:   A MSRuleCleaner workflow representation
+        :return:         The workflow object
+        """
+        if wflow['InputDataset'] and wflow['IncludeParents']:
+            childDataset = wflow['InputDataset']
+            parentDataset = findParent([childDataset], self.msConfig['dbsUrl'])
+            # NOTE: If findParent() returned None then the DBS service failed to
+            #       resolve the request (it is considered an ERROR outside WMCore)
+            if parentDataset is None:
+                msg = "Failed to resolve parent dataset for: %s in wflow: %s" % (childDataset, wflow['RequestName'])
```

Comment:
I would suggest to write the `wflow` word as a whole, thus `workflow`.

Reply:
Done

Reply:
Done